.travis.yml

```diff
@@ -1,27 +1,98 @@
 language: rust
+sudo: false
 branches:
   only:
     - master
-os:
-  - linux
-  - windows
-rust:
-  - stable
-  - beta
-  - nightly
+
 # Need to cache the whole `.cargo` directory to keep .crates.toml for
 # cargo-update to work, but don't cache the cargo registry
 before_cache:
   - rm -rf /home/travis/.cargo/registry
 cache:
   directories:
     - /home/travis/.cargo
-before_script:
-  - if [[ "$TRAVIS_RUST_VERSION" == "stable" || "$TRAVIS_RUST_VERSION" == "nightly" ]]; then rustup component add clippy-preview; fi
-script:
-  - if [[ "$TRAVIS_RUST_VERSION" == "stable" || "$TRAVIS_RUST_VERSION" == "nightly" ]]; then cargo clippy --all-targets --all-features -- -D warnings; fi
-  - cargo test --verbose
+
+DEPLOY_TO_GITHUB: &DEPLOY_TO_GITHUB
+  before_deploy:
+    - git config --local user.name "Stuart Crust"
+    - git config --local user.email "stuart.crust@gmail.com"
+    - name="supernova-$TRAVIS_TAG-$TARGET"
+    - mkdir $name
+    - cp target/$TARGET/release/supernova $name/
+    - cp README.md LICENSE-MIT LICENSE-APACHE supernova.sha1 $name/
+    - tar czvf $name.tar.gz $name
+  deploy:
+    provider: releases
+    api_key: $GH_TOKEN
+    file: supernova-$TRAVIS_TAG-$TARGET.tar.gz
+    skip_cleanup: true
+    on:
+      branch: master
+      repo: 0xazure/supernova
+      tags: true
+
 matrix:
   allow_failures:
     - rust: nightly
   fast_finish: true
+  include:
+    - name: Linux-Stable
+      env: TARGET=x86_64-unknown-linux-gnu
+      rust: stable
+      before_script:
+        - rustup component add clippy-preview
+      script:
+        - cargo clippy --all-targets --all-features -- -D warnings
+        - cargo test --verbose
+        - cargo build --release --target $TARGET --locked
+        - sha1sum target/$TARGET/release/supernova > supernova.sha1
+      <<: *DEPLOY_TO_GITHUB
+
+    - name: Linux-Beta
+      env: TARGET=x86_64-unknown-linux-gnu
+      rust: beta
+      script:
+        - cargo test --verbose
+        - cargo build --release
```

Comment:
Do we need the `--release` flag on release channels that aren't going to be built and uploaded? Travis' default `script` is `cargo build --verbose; cargo test --verbose`, `--release` adds optimization cycles we may not need here.

Reply:
Ah good catch - will get rid of them